components/ILIAS/LDAP/classes/LDAPRoleMappingTable.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ ********************************************************************
+ */
+
+declare(strict_types=1);
+
+use ILIAS\Data\Order;
+use ILIAS\Data\Range;
+use ILIAS\UI\URLBuilder;
+use ILIAS\UI\URLBuilderToken;
+use ILIAS\UI\Factory as UIFactory;
+use ILIAS\Data\Factory as DataFactory;
+use ILIAS\UI\Component\Table\Column\Column;
+use ILIAS\UI\Component\Table\DataRetrieval;
+use Psr\Http\Message\ServerRequestInterface;
+use ILIAS\UI\Component\Table\Data as DataTable;
+use ILIAS\UI\Implementation\Component\Table\Action\Action;
+
+class LDAPRoleMappingTable implements DataRetrieval
+{
+    /** @var array[][]|null */
+    private ?array $records = null;
```

Comment:
Could we specify the shape of the array (`list<array{id: int, title: string, ...}>`), similar to `getRecords`?